addons/website/static/src/components/configurator/configurator.js

```diff
@@ -51,6 +53,9 @@ const PALETTE_NAMES = [
     'enark-4',
 ];
 
+const SUPPORTED_IMAGE_MIMETYPES = ['image/gif', 'image/jpe', 'image/jpeg', 'image/jpg', 'image/png', 'image/svg+xml'];
+const SUPPORTED_IMAGE_EXTENSIONS = ['.gif', '.jpe', '.jpeg', '.jpg', '.png', '.svg'];
```

Comment:
Indeed the logo image is really uploaded only once the configurator is fully completed. So we can't go through the `/web_editor/attachment/add_data` route since it creates a new attachment.

Reply:
If that's the only reason, I would have preferred that to duplicated code (I prefer anything over duplicated code really)... It's the same in the editor: if you upload something that you will not use, it will be created for no reason. In the case of logo upload, it is far less critical than in the editor (people have a logo or not). Anyways, let's keep it like that.

Reply:
Maybe the `web_editor/attachment/add_data` route could take a new parameter `check` that would not create anything but would just check if the image is supported (so basically everything except the attachment.create part) and would return True or the error, could probably be useful later for another implementation (if it was needed here, surely it might be needed elsewhere too)

Reply:
Mmh, would be strange to have a route call add_data that just checks, no? I would really rather just call add_data normally here:
- The user uploaded something wrong -> fails directly (not waiting for the last step of the editor to know that the logo is wrong and cannot be used)
- The user uploaded something good -> Best create the attachment directly, it's more than probably the logo the user wants, creating the attachment now will decrease website loading after last step.
- The user uploaded something good but it is not what he wanted (what the hell is he doing?) -> Worst case, the image appears in the media manager, he can delete it during its first edition if needed.

Anyways, I am fine with everything here.

Reply:
As long as this avoid duplication in JS, and it actually reach our endpoint that will actually do the real check about the image (add_data controller) I think this is good :+1: 
What you described makes sense

Reply:
Ok looks good to me too

Reply:
When we build the webiste at the end of the configurator (call to `configurator_apply`) we receive the logo in base64 from the configurator and set it on the website logo field:
![Screenshot from 2021-08-31 11-46-33](https://user-images.githubusercontent.com/49784121/131483464-246a0aa8-28f2-46e3-95f9-9d1f04e9df4c.png)

I thought I could get ride of this part by overwriting the attachment linked to the website logo but the `add_data` controller just allows to create new attachments. So the only optimization which can be done is: instead of passing the whole logo in the RPC call, just give the attachment id. Then `configurator_apply` could use this id to retrieve the attachment and set the logo. It would avoid two logo uploads and it would actually use the created attachment. In this version of the code the `add_data` controller create unused attachments...